ts/features/design-system/navigation/params.ts

```diff
@@ -0,0 +1,24 @@
+import DESIGN_SYSTEM_ROUTES from "./routes";
+
+type DesignSystemRoute = {
+  route: string;
+  title: string;
+};
+
+export type DesignSystemParamsList = {
+  [DESIGN_SYSTEM_ROUTES.MAIN]: undefined;
+  [DESIGN_SYSTEM_ROUTES.FOUNDATION.COLOR.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.FOUNDATION.TYPOGRAPHY.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.FOUNDATION.ICONS.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.FOUNDATION.PICTOGRAMS.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.FOUNDATION.LOGOS.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.COMPONENTS.BUTTONS.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.COMPONENTS.SELECTION.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.COMPONENTS.ADVICE.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.COMPONENTS.ACCORDION.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.COMPONENTS.TEXT_FIELDS.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.COMPONENTS.LIST_ITEMS.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.COMPONENTS.TOASTS.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.LEGACY.PICTOGRAMS.route]: DesignSystemRoute;
+  [DESIGN_SYSTEM_ROUTES.LEGACY.ILLUSTRATIONS.route]: DesignSystemRoute;
```

Comment:
all of these param types can be undefined since there's no usage of navigation params inside the related screens, do you agree?
```suggestion
  [DESIGN_SYSTEM_ROUTES.FOUNDATION.COLOR.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.FOUNDATION.TYPOGRAPHY.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.FOUNDATION.ICONS.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.FOUNDATION.PICTOGRAMS.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.FOUNDATION.LOGOS.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.COMPONENTS.BUTTONS.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.COMPONENTS.SELECTION.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.COMPONENTS.ADVICE.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.COMPONENTS.ACCORDION.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.COMPONENTS.TEXT_FIELDS.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.COMPONENTS.LIST_ITEMS.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.COMPONENTS.TOASTS.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.LEGACY.PICTOGRAMS.route]: undefined;
  [DESIGN_SYSTEM_ROUTES.LEGACY.ILLUSTRATIONS.route]: undefined;
```